massa-ledger/src/ledger.rs

```diff
@@ -150,9 +150,11 @@ impl FinalLedger {
         self.sorted_ledger
             .get_entry(addr, LedgerSubEntry::Balance)
             .map(|bytes| {
-                Amount::from_raw(u64::from_be_bytes(
-                    bytes.try_into().expect("critical: invalid balance format"),
-                ))
+                Amount::from_raw(
```

Comment:
Amount should have its own compact deserialization routine (so that from_raw is never used)

Reply:
This particular one was not supposed to stay, what I'm doing now is: 
```rust
    pub fn get_parallel_balance(&self, addr: &Address) -> Option<Amount> {
        self.sorted_ledger
            .get_entry(addr, LedgerSubEntry::Balance)
            .map(|bytes| {
                Amount::from_raw(
                    u64::from_varint_bytes(&bytes)
                        .expect("critical: invalid balance format")
                        .0,
                )
            })
    }
```

Is that how it should be done?

Reply:
from_raw / to_raw should basically never be used (except when we absolutely need it in the SC engine)

This code snippet assumes that u64 is the underlying representation of Amount, which might change.

We should always use Amount::from_bytes_compact and Amount::to_bytes_compact for serialization, and never assume anything about the underlying representation of Amount outside of the Amount module.

Reply:
Oh I forgot there was an implemented trait that already does that, will change it now.

Reply:
I join here to say that now in main there is the new way to deserialize for amount also with the deserializer. Maybe use it to avoid change in the near future.